axelor-human-resource/src/main/java/com/axelor/apps/hr/service/timesheet/TimesheetServiceImpl.java

```diff
@@ -1218,6 +1223,97 @@ protected TimesheetLine createTimeSheetLineFromPPT(
     return timesheetLine;
   }
 
+  protected BigDecimal getConvertedPlannedTime(
+      Timesheet timesheet, ProjectPlanningTime projectPlanningTime) throws AxelorException {
+
+    String timeLoggingPreference = timesheet.getTimeLoggingPreferenceSelect();
+
+    if (timeLoggingPreference == null) {
+      timeLoggingPreference = timesheet.getEmployee().getTimeLoggingPreferenceSelect();
+    }
+
+    switch (timeLoggingPreference) {
+      case EmployeeRepository.TIME_PREFERENCE_DAYS:
+        return computeDaysDuration(timesheet, projectPlanningTime);
+      case EmployeeRepository.TIME_PREFERENCE_HOURS:
+        return computeHoursDuration(timesheet, projectPlanningTime);
+      case EmployeeRepository.TIME_PREFERENCE_MINUTES:
+        return computeMinutesDuration(timesheet, projectPlanningTime);
+      default:
+        return BigDecimal.ZERO;
+    }
+  }
+
+  public BigDecimal computeDaysDuration(
+      Timesheet timesheet, ProjectPlanningTime projectPlanningTime) throws AxelorException {
+    AppBase appBase = appBaseService.getAppBase();
+    BigDecimal dailyWorkHours = timesheet.getEmployee().getDailyWorkHours();
+    BigDecimal time = projectPlanningTime.getPlannedTime();
+    Unit timeUnit = projectPlanningTime.getTimeUnit();
+
+    if (dailyWorkHours == null || dailyWorkHours.compareTo(BigDecimal.ZERO) == 0) {
+      throw new AxelorException(
+          TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
+          I18n.get(HumanResourceExceptionMessage.TIMESHEET_DAILY_WORK_HOURS));
+    }
+
+    if (timeUnit.equals(appBase.getUnitDays())) {
+      return time;
+    } else if (timeUnit.equals(appBase.getUnitHours())) {
+      return time.divide(dailyWorkHours, 2, RoundingMode.HALF_DOWN);
```

Comment:
is there a specific reason to use HALF_DOWN ? if not please use HALF_UP instead everywhere